mlir/lib/Dialect/Tosa/IR/TosaCanonicalizations.cpp

```diff
@@ -771,7 +771,7 @@ OpFoldResult ConstOp::fold(FoldAdaptor adaptor) { return getValueAttr(); }
     ShapedType inputTy = llvm::cast<ShapedType>(getInput().getType());         \
     if (!inputTy.hasRank())                                                    \
       return {};                                                               \
-    if (inputTy.getDimSize(getAxis()) == 1)                                    \
+    if (inputTy.getRank() != 0 && inputTy.getDimSize(getAxis()) == 1)          \
```

Comment:
Shouldn't reduction on rank0 tensor get folded as well?

Reply:
Would you like me to change it to `if (inputTy.getRank() == 0 || inputTy.getDimSize(getAxis()) == 1)` ? That would solve the issue and fold rank0 tensors as well.